src/marqo/tensor_search/tensor_search.py

```diff
@@ -1341,65 +1337,56 @@ def construct_vector_input_batches(query: Union[str, Dict], index_info) -> Tuple
             return [k for k, _ in ordered_queries], []
 
 
-def get_vector_properties_to_search(searchable_attributes: Union[None, List[str]], index_info: IndexInfo) -> List[str]:
+def get_vector_properties_to_search(searchable_attributes: Union[None, List[str]], index_info: IndexInfo, offset: int = 0) -> List[str]:
     if searchable_attributes is None:
-        return index_info.get_vector_properties().keys()
+        properties_to_search = index_info.get_vector_properties().keys()
     else:
         searchable_attributes_as_vectors = {
             utils.generate_vector_name(field_name=attribute) for attribute in searchable_attributes
         }
         # discard searchable attributes that aren't found in the cache:
-        return list(searchable_attributes_as_vectors.intersection(
+        properties_to_search = list(searchable_attributes_as_vectors.intersection(
             index_info.get_vector_properties().keys()
         ))
+    
+    # Validation for offset (pagination is single field) if offset not provided, validation is not run.
+    if len(properties_to_search) != 1 and offset > 0:
+        human_readable_vector_properties = [v.replace(TensorField.vector_prefix, '') for v in
+                                            list(properties_to_search)]
+        raise errors.InvalidArgError(
+            f"Pagination (offset > 0) is only supported for single field searches!"
+            f" Your search currently has {len(properties_to_search)} vectorisable fields: {human_readable_vector_properties}"
+        )
+    return properties_to_search
 
 
-def construct_msearch_body_elements(vector_properties_to_search: List[str], offset: int, filter_string: str, index_info: IndexInfo, result_count: int, query_vector: List[float], attributes_to_retrieve: List[str], index_name: str, contextualised_filter: str) -> List[Dict[str, Any]]:
+def construct_msearch_body_elements(searchableAttributes: List[str], offset: int, filter_string: str, index_info: IndexInfo, result_count: int, query_vector: List[float], attributes_to_retrieve: List[str], index_name: str, score_modifiers: Optional[Dict[str, Any]] = None) -> List[Dict[str, Any]]:
```

Comment:
This change lets:
1. search use `construct_msearch_body_elements`
2. Bulk search to use score modifiers